### DISCUSSION TITLE
New Optimization algorithm: Nelder Mead Simplex

### DISCUSSION THREAD

**Discussion by CivilizationalAgency** [akumidv/tradingview-assistant-chrome-extension#191] [Ideas]
Would be nice to have something more sophisticated than hillclimbing ("sequential improvement") that can update multiple parameters at each step, and that also converges faster than annealing (and with fewer hyperparameters)

**Comment by vindiou** [comment]
| Feature/Metric           | Nelder-Mead Simplex | Genetic Optimization   | Simulated Annealing   |
|--------------------------|---------------------|-------------------------|-----------------------|
| **Speed (low-dimension)**| Fast               | Slower                 | Moderate             |
| **Speed (high-dimension)**| Poor scaling       | Moderate (parallelizable)| Slow                |
| **Global Optimization**  | Weak               | Strong                 | Moderate             |
| **Local Optimization**   | Strong             | Moderate               | Weak to Moderate     |
| **Robustness**           | Low                | High                   | Moderate             |
| **Parallelization**      | Difficult          | Easy                   | Difficult            |
| **Complex Problems**     | Poor performance   | Excellent              | Good                 |

**Comment by akumidv** [comment]
Gradient boosting I think better, but all this algorithms have there weeks. At least how to get stuck in local areas and probability to found better zone.
For strategy optimization local zone optimization is not valuable, more important to found best zone itself and annealing is gold middle. But I have in plans implement another algorithms also a implementing parallelization on different accounts. In new version.
